src/simulation/simulation.jl

```diff
@@ -639,19 +639,17 @@ function _execute!(
 
                 TimerOutputs.@timeit RUN_SIMULATION_TIMER "Solve $(model_name)" begin
                     settings = get_settings(model)
+                    @show model_name
                     status =
                         solve!(step, model, get_current_time(sim), store; exports = exports)
-                    if get_allow_fails(settings) && (status != RunStatus.SUCCESSFUL)
-                        continue
-                    elseif !get_allow_fails(settings) && (status != RunStatus.SUCCESSFUL)
-                        throw(
-                            ErrorException(
-                                "Simulation Failed in problem $(model_name). Returned $(status)",
-                            ),
-                        )
-                    else
-                        @assert status == RunStatus.SUCCESSFUL
-                    end
+                    # TODO: This block of code is currently not in the execution path after a failed solve
```

Comment:
@daniel-thom The current way we have the try-catch and loggers, etc make it such that if the simulation fails it never uses this block of code we need to address that.

Reply:
The simplest solution is to wrap this in a try-catch. I'm assuming that you want to keep the current throw logic in solve.

Reply:
We need to be careful not to wrap the time critical code in try-catch